modules/auxiliary/scanner/http/nagios_xi_scanner.rb

```diff
@@ -77,9 +77,48 @@ def login_after_install_or_license
 
   end
 
+  def parse_version(nagios_version)
+    # strip the xi- prefix used for Nagios XI versions in the official documentation
+    clean_version = nagios_version.downcase.delete_prefix('xi-')
+
+    # check for special case of 5r1.0, which is actually 5.1.0 but for some reason they still added the `r` used in legacy versions (this is the only version with this format)
+    if clean_version == '5r1.0'
+      clean_version = '5.1.0'
+    end
+
+    # Since we are dealing with a user provided version here, a strict regex check seems appropriate
+    # So far all modern Nagios XI versions start with 5. This regex can be updated if Nagios XI 6 ever comes out and modules for this new version are added
+    modern_check = clean_version.scan(/5\.\d{1,2}\.\d{1,2}/).flatten.first
+
+    unless modern_check.blank?
+      return ['modern', clean_version]
+    end
+
+    # Nagios XI legacy version numbers vary a lot, but they always start with a year followed by `r` and a single digit
+    legacy_check = clean_version.scan(/(20\d{2}r\d{1})/).flatten.first
```

Comment:
While it's true that the version is followed by "a single digit" this will result in incorrect version matching.

The `(20\d{2}r\d{1})` match group will truncate anything after the first digit after `r` which is quite common.

Fortunately, they're all released within the same year, so the disparity isn't too large. Given that the software is already quite old I wouldn't worry about it.

* xi-2014r2.7
* xi-2014r2.6
* xi-2014r2.5
* xi-2014r2.4
* xi-2014r2.3
* xi-2014r2.2
* xi-2014r2.1
* xi-2014r2.0
* xi-2014r1.5
* xi-2014r1.4
* xi-2014r1.3
* xi-2014r1.2
* xi-2014r1.1
* xi-2014r1.0
* xi-2014r1.0RC3
* xi-2014r1.0RC2
* xi-2014r1.0RC1
* xi-2012r2.9
* xi-2012r2.8c
* xi-2012r2.8b
* xi-2012r2.8
* xi-2012r2.7
* xi-2012r2.6
* xi-2012r2.5
* xi-2012r2.4b
* xi-2012r2.4
* xi-2012r2.3
* xi-2012r2.2
* xi-2012r2.1
* xi-2012r2.0
* xi-2012r1.8
* xi-2012r1.7
* xi-2012r1.6
* xi-2012r1.5b
* xi-2012r1.5
* xi-2012r1.4
* xi-2012r1.3
* xi-2012r1.2
* xi-2012r1.1
* xi-2012r1.0
* xi-2011r3.3
* xi-2011r3.2
* xi-2011r3.1
* xi-2011r3.0
* xi-2011r2.4
* xi-2011r2.3
* xi-2011r2.2
* xi-2011r2.1
* xi-2011r2.0
* xi-2011r1.9
* xi-2011r1.8
* xi-2011r1.7
* xi-2009r1.4bi
* xi-2011r1.6
* xi-2011r1.5
* xi-2011r1.3
* xi-2011r1.2
* xi-2011r1.1
* xi-2011r1
* xi-2009r1.4b
* xi-2009r1.4
* xi-2009r1.3g
* xi-2009r1.3f
* xi-2009r1.3e
* xi-2009r1.3d
* xi-2009r1.3c
* xi-2009r1.3b
* xi-2009r1.3
* xi-2009r1.2d
* xi-2009r1.2c
* xi-2009r1.2
* xi-2009r1.1h
* xi-2009r1.1g
* xi-2009r1.1f
* xi-2009r1.1e
* xi-2009r1.1d
* xi-2009r1.1c
* xi-2009r1.1b
* xi-2009r1fix
* xi-2009r1
* xi-2009r1.1a